File: macos/Classes/RCacheFlutterMethod.swift
//
//  RCacheFlutterMethod.swift
//  rcache_flutter
//
//  Created by Rahmat Trinanda Pramudya Amar on 24/08/24.
//

import Foundation

internal enum RCacheFlutterMethod: String {
    case channel = "id.nesd.rcache_flutter"
    
    internal enum Key: String {
        case common = "common"
        case credentials = "credentials"
        case remove = "remove"
        case clear = "clear"
        
        internal enum Save: String {
            case uint8List = "saveUint8List"
            case string = "saveString"
            case bool = "saveBool"
            case integer = "saveInteger"
            case array = "saveArray"
            case map = "saveMap"
            case double = "saveDouble"
            
        }
        
        internal enum Read: String {
            case uint8List = "readUint8List"
            case string = "readString"
            case bool = "readBool"
            case integer = "readInteger"
            case array = "readArray"
            case map = "readMap"
            case double = "readDouble"
        }
    }
}

File: android/src/main/kotlin/id/nesd/rcache_flutter/RCacheFlutterMethod.kt
package id.nesd.rcache_flutter

internal class RCacheFlutterMethod {

    companion object {
        const val channel: String = "id.nesd.rcache_flutter"
    }

    class Key {

        companion object {
            const val common: String = "common"
            const val credentials: String = "credentials"
            const val remove: String = "remove"
            const val clear: String = "clear"
        }

        class Save {
            companion object {
                const val uint8List: String = "saveUint8List"
                const val string: String = "saveString"
                const val bool: String = "saveBool"
                const val integer: String = "saveInteger"
                const val array: String = "saveArray"
                const val map: String = "saveMap"
                const val double: String = "saveDouble"
            }
        }

        class Read {
            companion object {
                const val uint8List: String = "readUint8List"
                const val string: String = "readString"
                const val bool: String = "readBool"
                const val integer: String = "readInteger"
                const val array: String = "readArray"
                const val map: String = "readMap"
                const val double: String = "readDouble"
            }
        }
    }
}